### DISCUSSION TITLE
Is "THREADS" option working on pyswmm?

### DISCUSSION THREAD

**Discussion by sausagechen** [pyswmm/pyswmm#503] [Q&A] [answered]
In SWMM inp file, the THREADS in [OPTIONS] help the model running faster in the software. But running the same inp file with pyswmm cannot boost. I'm using Pycharm. 
I'd like to know whether this is normal and how to overcome?

**Comment by bemcdonnell** [comment]
@sausagechen how are you confirming if it works? Are you looking at the CPU usage?

**Reply by sausagechen** [reply]
Exactly, by Taskmgr.exe.

**Reply by bemcdonnell** [reply]
@sausagechen do you mind running a few tests on this to confirm it’s a problem? 

1. Set the threads inside the options section of the model to 8 and run in. At the footer of the RPT file, check the time it took to run the model. 
2. Use this same model but before the simulation starts, use pyswmm to put threads to 1. Check the RPT again
3. Then set the threads in the options section to 1, run it
4. Use this same model and use pyswmm to set the threads to 8.

**Comment by sausagechen** [comment, marked as answer]
After comparing the methods you've posted, I've noticed that it doesn't seem to reproduce the problem I mentioned before when just running the model. It's possible that there is a time lag from other code that seems to be taking a lot of time in traversing the objects in Nodes(sim).

**Comment by sausagechen** [comment]
I would like to record the values of multiple pollutants in each step, but when I open the Output object, it doesn't seem to be able to derive the appropriate dictionary, only an associated variable named POLLUT_CONC_0. This results in me needing to traverse the Nodes(sim) object multiple times in the run.
Can the above be optimized please?
